BlackMarket/Managers/SessionManager.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Constaints.swift
+//  swiftui-base
+//
+//  Created by Karen Stoletniy on 06/9/22.
+//
+
+import SwiftUI
+import Combine
+
+internal final class SessionManager {
+  
+  var isSessionValidPublisher: AnyPublisher<Bool, Never> {
+    currentSessionTokenPublisher
+      .filter { _ in self.publishTokenState }
+      .map { $0 != nil }
+      .eraseToAnyPublisher()
+  }
+  
+  private var currentSessionTokenPublisher: AnyPublisher<String?, Never> {
+    userDefaults.publisher(for: \.currentSessionToken).eraseToAnyPublisher()
+  }
+  
+  private var subscriptions = Set<AnyCancellable>()
+  
+  static let shared = SessionManager()
+  
+  private let userDefaults: UserDefaults
+  private var publishTokenState: Bool = true
+  
+  init(userDefaults: UserDefaults = .standard) {
+    self.userDefaults = userDefaults
+  }
+  
+  private(set) var accessToken: String? {
+    get {
+      userDefaults.currentSessionToken
+    }
+    set {
+      userDefaults.currentSessionToken = newValue
+    }
+  }
+  
+  @MainActor
+  func saveSession(withToken token: String?, publishState: Bool = true) {
+    self.publishTokenState = publishState
+    accessToken = token
+  }
+  
+  func deleteSession(publishState: Bool = true) {
+    self.publishTokenState = publishState
+    accessToken = nil
+  }
+  
+  func getHeaders() -> [String: String] {
+    return [
+      "Authorization": "bearer \(accessToken ?? "")",
+      "x-platform": "ios"
```

Comment:
Should  the session manager in charge of creating these headers? Seems like a responsibility for the APIClient or the headers provider for such APIClient